soak/README.md

```diff
@@ -0,0 +1,202 @@
+## INTRODUCTION
+
+ACK team provides the soak testing framework for service teams where service teams can execute long running
+soak tests for testing the performance and stability of the service controllers.
+
+Under the hood, these soak tests consist of multiple end to end test runs for the service controller. End to end tests are present in service-controller github repository.
+i.e. "https://github.com/aws-controllers-k8s/<service-name>-controller".
+
+If you wish to run soak tests for your service controller locally, see the "HOW TO RUN SOAK TEST LOCALLY" section below.
+
+## RUNNING SOAK TESTS USING ACK PROW JOBS
+
+TBD
+
+
+## HOW TO RUN SOAK TEST LOCALLY
+
+TODO
+```
+* Convert following 9 steps into a single executable script with default ack soak-test params. 
+* Service teams will have the freedom to set parameters and customizing their soak-test runs.
+```
+
+### Prerequisites
+* A tool for building OCI images (docker, buildah etc..)
+* A kubernetes cluster to run soak tests
+* helm 
+* kubectl
+
+
+### STEP 1 (Install the ACK service controller using helm)
+
+In this step we install ack service controller which will be soak tested using the soak-test runner created in step 4.
+
+You can install an ACK controller following the documentation [here](https://aws-controllers-k8s.github.io/community/user-docs/install/)
+but following commands are also TL;DR for the installation documentation.
+
+You can use either (a) or (b) method below.
+
+a) Pull the chart from public repo
+
+* RUN `export SERVICE_NAME=<aws-service-name>` Ex: 'apigatewayv2'
+* RUN `export CONTROLLER_VERSION=<semver>` Ex: v0.0.2
+* RUN `export CONTROLLER_CHART_URL=public.ecr.aws/aws-controllers-k8s/chart:$SERVICE_NAME-$CONTROLLER_VERSION`
+* RUN `export CONTROLLER_CHART_RELEASE_NAME=<controller-chart-release-name>` Ex: soak-test
+* RUN `helm chart pull $CONTROLLER_CHART_URL`
+* RUN `helm chart install $CONTROLLER_CHART_RELEASE_NAME $CONTROLLER_CHART_URL` 
+
+b) Use the chart from source repository
+
+* RUN `export SERVICE_NAME=<aws-service-name>` Ex: 'apigatewayv2'
+* checkout `<aws-service-name>-controller` github repository
+* RUN `cd helm`
+* RUN `export CONTROLLER_CHART_RELEASE_NAME=<controller-chart-release-name>` Ex: soak-test
+* RUN `helm install $CONTROLLER_CHART_RELEASE_NAME .` Note: Set custom values during helm install
+
+NOTE
+* Use "--set serviceAccount.Annotations.eks.amazonaws.com/role-arn=<IAM_ROLE_ARN_FOR_IRSA>" during helm install.
```

Comment:
I would include this as mandatory in the above block - temporary credentials cannot be used because they would expire within 24hrs.